helix-vcs/src/git.rs

```diff
@@ -87,6 +241,90 @@ impl DiffProvider for Git {
         }
         Some(data)
     }
+
+    fn get_changed_files(&self, cwd: &Path) -> Result<Vec<FileChange>> {
+        Self::status(
+            &Self::open_repo(cwd, None)
+                .ok_or_else(|| anyhow::anyhow!("no Git repository found"))?
+                .to_thread_local(),
+        )
+    }
+}
+
+impl RawChanges {
+    pub fn add_addition(&mut self, addition: RawAddition) {
+        self.additions.push(addition);
+    }
+
+    pub fn add_deletion(&mut self, deletion: RawDeletion) {
+        match self.deletions.entry(deletion.oid) {
+            Entry::Occupied(entry) => {
+                entry.into_mut().push(deletion);
+            }
+            Entry::Vacant(entry) => {
+                entry.insert(vec![deletion]);
+            }
+        }
+    }
+
+    pub fn add_modification(&mut self, modification: RawModification) {
+        self.modifications.push(modification);
+    }
+}
+
+impl From<RawChanges> for Vec<FileChange> {
+    // Unlike Git, we only look for pure renames at the moment.
+    // TODO: detect renames with minor changes
+    fn from(mut raw: RawChanges) -> Self {
+        let mut status_entries = vec![];
+
+        let additions_left = if !raw.additions.is_empty() && !raw.deletions.is_empty() {
+            let mut unmatched_additions = vec![];
+
+            for add in raw.additions.into_iter() {
+                let matched_deletions = match raw.deletions.entry(add.oid) {
+                    Entry::Occupied(entry) => entry.into_mut(),
+                    Entry::Vacant(_) => {
+                        unmatched_additions.push(add);
+                        continue;
+                    }
+                };
+
+                // Impossible to have an empty vec inside
+                let chosen_deletion = matched_deletions.pop().expect("unexpected empty vec");
+                if matched_deletions.is_empty() {
+                    raw.deletions.remove(&add.oid);
+                }
+
+                status_entries.push(FileChange::Renamed {
+                    from_path: chosen_deletion.path.to_owned(),
+                    to_path: add.path.to_owned(),
+                });
+            }
+
+            unmatched_additions
+        } else {
+            raw.additions
+        };
+
+        additions_left
+            .into_iter()
+            .for_each(|item| status_entries.push(FileChange::Untracked { path: item.path }));
+        raw.deletions
+            .values()
+            .into_iter()
+            .flat_map(|val| val.iter())
+            .for_each(|item| {
+                status_entries.push(FileChange::Deleted {
+                    path: item.path.to_owned(),
+                })
+            });
+        raw.modifications
+            .into_iter()
+            .for_each(|item| status_entries.push(FileChange::Modified { path: item.path }));
```

Comment:
```suggestion
let status_entries = additions_left
    .into_iter()
    .map(|item| FileChange::Untracked { path: item.path })
    .chain(
        raw.deletions
            .values()
            .into_iter()
            .flat_map(|val| val.iter())
            .map(|item| {
                FileChange::Deleted {
                    path: item.path.to_owned(),
                }
            })
    )
    .chain(
        raw.modifications
            .into_iter()
            .map(|item| FileChange::Modified { path: item.path })
    )
     .collect::<Vec<_>>();
```

Reply:
Thanks for the suggestion but it looks like the suggested code isn't formatted? If I commit the suggestion CI will turn red I think :/

Reply:
It's fine. I will accept it on GitHub and amend locally :)